### PR TITLE
core: added labels to rook v1

### PR DESCRIPTION
**Pull request by galexrt**
**Description of your changes:**

This adds Labels and LabelsSpec to the rook.io/v1 apis pkg.

Related to #6050

Signed-off-by: Alexander Trost <galexrt@googlemail.com>

**Which issue is resolved by this Pull Request:**

Related to #6050

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [x] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [x] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [x] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [x] Documentation has been updated, if necessary.
- [x] Unit tests have been added, if necessary.
- [x] Integration tests have been added, if necessary.
- [x] Pending release notes updated with breaking and/or notable changes, if necessary.
- [x] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [x] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

**Comment by galexrt** [comment]
Restarting Jenkins (all green except `aws_1.15.x`).

**Comment by galexrt** [comment]
@leseb The CRDs of Cassandra, Ceph, etc. are not changed in this PR.

The upcoming PRs, e.g., #6084, are the actual implementations of `labels` in the CRD for each each storage backend (one by one as of right now) and that is where I'm going to update the docs + examples.

File: pkg/apis/rook.io/v1/labels.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	for k, v := range a {
+		if _, ok := t.Labels[k]; !ok {
+			t.Labels[k] = v
+		}
+	}
+}
+
+// Merge returns an Labels which results from merging the attributes of the
+// original Labels with the attributes of the supplied one. The supplied
+// Placement's attributes will override the original ones if defined.
```
Comment (leseb) [inline pkg/apis/rook.io/v1/labels.go:41]:
```suggestion
// Labels attributes will override the original ones if defined.
```

?

**Comment by galexrt** [comment]
@travisn Got it, I have the Ceph implementation already lined up in #6084.

**Pull request merged**